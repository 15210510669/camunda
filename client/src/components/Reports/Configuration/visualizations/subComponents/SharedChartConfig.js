import React from 'react';
import {ColorPicker, Switch, LabeledInput} from 'components';
import RelativeAbsoluteSelection from './RelativeAbsoluteSelection';
import ChartTargetInput from './ChartTargetInput';
import './SharedChartConfig.scss';

export default function SharedChartConfig({configuration, onChange, report, hideTooltipOptions}) {
  const enabledTarget = configuration.targetValue ? configuration.targetValue.active : false;
  const defaultValues = {target: '', dateFormat: '', isBelow: false};
  const isSingleReport = report.reportType === 'single';
  const isFrequency = isSingleReport
    ? report.data.view.property === 'frequency'
    : Object.values(report.result)[0].data.view.property === 'frequency';

  return (
    <div className="SharedChartConfig">
      {isSingleReport && (
        <fieldset className="ColorSection">
          <legend>Select visualization color</legend>
          <ColorPicker selectedColor={configuration.color} onChange={onChange} />
        </fieldset>
      )}
      {!hideTooltipOptions && (
        <fieldset className="tooltipOptions">
          <legend>Tooltips options</legend>
          <RelativeAbsoluteSelection
            relativeDisabled={!isFrequency}
            configuration={configuration}
            onChange={onChange}
          />
        </fieldset>
      )}
      <fieldset className="axisConfig">
        <legend>Axis names</legend>
        <LabeledInput
          label="x-axis"
          type="text"
          value={configuration.xLabel}
          onChange={({target: {value}}) => onChange('xLabel', value)}
        />
        <LabeledInput
          label="y-axis"
          type="text"
          value={configuration.yLabel}
          onChange={({target: {value}}) => onChange('yLabel', value)}
        />
      </fieldset>
      <fieldset className="goalLine" disabled={!enabledTarget}>
        <legend>
          <Switch
            checked={enabledTarget}
            onChange={({target: {checked}}) =>
              onChange('targetValue', {
                values: defaultValues,
                ...configuration.targetValue,
                active: checked
              })
            }
          />
          Goal
        </legend>
        <ChartTargetInput {...{configuration, onChange, report}} />
      </fieldset>
    </div>
  );
}
