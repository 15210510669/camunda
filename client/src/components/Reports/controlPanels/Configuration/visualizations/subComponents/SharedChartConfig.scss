.SharedChartConfig fieldset {
  margin: 5px 0;

  &.goalLine[disabled] .ChartTargetInput,
  &.tooltipOptions[disabled] {
    opacity: 0.4;
    pointer-events: none;
  }
}
