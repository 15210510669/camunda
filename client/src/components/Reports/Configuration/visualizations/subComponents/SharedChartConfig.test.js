import React from 'react';
import {shallow} from 'enzyme';
import SharedChartConfig from './SharedChartConfig';

const report = {combined: false, data: {view: {property: 'frequency'}}};

const configuration = {
  showInstanceCount: false,
  color: '#1991c8',
  pointMarkers: true,
  hideRelativeValue: false,
  hideAbsoluteValue: false,
  xLabel: '',
  yLabel: '',
  targetValue: {active: false}
};

it('it should match snapshot - SharedChartConfig', () => {
  const node = shallow(<SharedChartConfig {...{report, configuration}} />);
  expect(node).toMatchSnapshot();
});

it('it should hide ColorPicker for combined report configuration', () => {
  const node = shallow(
    <SharedChartConfig
      {...{
        report: {
          combined: true,
          result: {
            test1: {data: {view: {property: 'frequency'}}}
          }
        },
        configuration
      }}
    />
  );

  expect(node.find('ColorPicker')).not.toBePresent();
});

it('it should hide tooltip options when specified in props', () => {
  const node = shallow(
    <SharedChartConfig
      {...{
        report,
        configuration,
        hideTooltipOptions: true
      }}
    />
  );

  expect(node.find('RelativeAbsoluteSelection')).not.toBePresent();
});

it('should invok onchange when changing xlabel/ylabel input on bar chart', () => {
  const spy = jest.fn();
  const node = shallow(<SharedChartConfig {...{report, configuration}} onChange={spy} />);

  node
    .find('LabeledInput')
    .first()
    .simulate('change', {target: {value: 'testLabel'}});

  expect(spy).toHaveBeenCalledWith('xLabel', 'testLabel');
});

it('should invok onchange when enabling target value switch', () => {
  const spy = jest.fn();
  const node = shallow(<SharedChartConfig {...{report, configuration}} onChange={spy} />);

  node.find('.goalLine Switch').simulate('change', {target: {checked: true}});

  expect(spy).toHaveBeenCalledWith('targetValue', {
    active: true,
    values: {dateFormat: '', isBelow: false, target: ''}
  });
});
