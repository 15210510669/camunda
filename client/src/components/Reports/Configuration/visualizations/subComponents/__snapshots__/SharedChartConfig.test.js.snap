// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should match snapshot - SharedChartConfig 1`] = `
<div
  className="SharedChartConfig"
>
  <fieldset
    className="ColorSection"
  >
    <legend>
      Select visualization color
    </legend>
    <ColorPicker
      colors={
        Array [
          "#B80000",
          "#DB3E00",
          "#FCCB00",
          "#008B02",
          "#006B76",
          "#1991c8",
          "#004DCF",
          "#5300EB",
          "#EB9694",
          "#FAD0C3",
          "#FEF3BD",
          "#C1E1C5",
          "#BEDADC",
          "#C4DEF6",
          "#BED3F3",
          "#D4C4FB",
        ]
      }
      selectedColor="#1991c8"
    />
  </fieldset>
  <fieldset
    className="tooltipOptions"
  >
    <legend>
      Tooltips options
    </legend>
    <RelativeAbsoluteSelection
      configuration={
        Object {
          "color": "#1991c8",
          "hideAbsoluteValue": false,
          "hideRelativeValue": false,
          "pointMarkers": true,
          "showInstanceCount": false,
          "targetValue": Object {
            "active": false,
          },
          "xLabel": "",
          "yLabel": "",
        }
      }
      relativeDisabled={false}
    />
  </fieldset>
  <fieldset
    className="axisConfig"
  >
    <legend>
      Axis names
    </legend>
    <LabeledInput
      label="x-axis"
      onChange={[Function]}
      type="text"
      value=""
    />
    <LabeledInput
      label="y-axis"
      onChange={[Function]}
      type="text"
      value=""
    />
  </fieldset>
  <fieldset
    className="goalLine"
    disabled={true}
  >
    <legend>
      <Switch
        checked={false}
        onChange={[Function]}
      />
      Goal
    </legend>
    <ChartTargetInput
      configuration={
        Object {
          "color": "#1991c8",
          "hideAbsoluteValue": false,
          "hideRelativeValue": false,
          "pointMarkers": true,
          "showInstanceCount": false,
          "targetValue": Object {
            "active": false,
          },
          "xLabel": "",
          "yLabel": "",
        }
      }
      report={
        Object {
          "combined": false,
          "data": Object {
            "view": Object {
              "property": "frequency",
            },
          },
        }
      }
    />
  </fieldset>
</div>
`;
